File: app/src/main/java/com/example/vladimir/easyenglishlearn/category_select/CategoryRemoveFragment.java
package com.example.vladimir.easyenglishlearn.category_select;

import androidx.lifecycle.ViewModelProvider;
import androidx.databinding.DataBindingUtil;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.DialogFragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.vladimir.easyenglishlearn.R;
import com.example.vladimir.easyenglishlearn.databinding.FragmentRemoveCategoryBinding;

import java.util.Objects;

import static com.example.vladimir.easyenglishlearn.Constants.CATEGORY_NAME;

public class CategoryRemoveFragment extends DialogFragment {

    private CategoryViewModel mViewModel;


    @NonNull
    public static DialogFragment newInstance(String categoryName) {
        Bundle args = new Bundle();
        args.putString(CATEGORY_NAME, categoryName);
        DialogFragment fragment = new CategoryRemoveFragment();
        fragment.setArguments(args);
        return fragment;
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater,
                             @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        FragmentRemoveCategoryBinding binding = DataBindingUtil.inflate(inflater,
                R.layout.fragment_remove_category,
                container,
                false);
        String categoryName = Objects.requireNonNull(getArguments()).getString(CATEGORY_NAME);
        binding.setCategoryName(categoryName);
        mViewModel = new ViewModelProvider(this).get(CategoryViewModel.class);
        binding.setViewModel(mViewModel);
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        mViewModel.getRemoveCategoryLiveData().observe(getViewLifecycleOwner(), aVoid -> closeDialog());
    }

    private void closeDialog() {
        dismiss();
    }
}

File: app/src/main/java/com/vladimir_x/easyenglishlearn/ui/category_select/CategoryRemoveFragment.kt
package com.vladimir_x.easyenglishlearn.ui.category_select

import android.os.Bundle
import android.view.View
import androidx.core.os.bundleOf
import androidx.fragment.app.DialogFragment
import com.vladimir_x.easyenglishlearn.Constants
import com.vladimir_x.easyenglishlearn.R
import com.vladimir_x.easyenglishlearn.databinding.FragmentRemoveCategoryBinding

class CategoryRemoveFragment : DialogFragment(R.layout.fragment_remove_category) {
    private var _binding: FragmentRemoveCategoryBinding? = null
    private val binding get() = _binding!!

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        _binding = FragmentRemoveCategoryBinding.bind(view)
        val categoryName = requireArguments().getString(Constants.CATEGORY_NAME) ?: ""
        binding.rcfTvRemoveCategory.text = categoryName
        binding.rcfBtnYes.setOnClickListener {
            parentFragmentManager.setFragmentResult(
                Constants.DIALOG_REMOVE_CATEGORY,
                bundleOf(Constants.RESULT_KEY to DialogResult.Yes)
            )
            closeDialog()
        }

        binding.rcfBtnNo.setOnClickListener {
            closeDialog()
        }
    }

    private fun closeDialog() {
        dismiss()
    }

    companion object {
        fun newInstance(categoryName: String?): DialogFragment {
            val args = Bundle()
            args.putString(Constants.CATEGORY_NAME, categoryName)
            val fragment: DialogFragment = CategoryRemoveFragment()
            fragment.arguments = args
            return fragment
        }
    }
}